src/screens/Blocked/elements/RenderList/index.js

```diff
@@ -1,167 +1,40 @@
-
-    const hanldeImage = (uri) => {
-      if(uri && typeof uri === 'string') {
-        return (
-          <Image
-              source={{
-                uri,
-              }}
-              style={styles.profilepicture}
-              width={48}
-              height={48}
-              testID='images'
-            /> 
-        )
-      }
-      return (
-        <View testID='noimage' style={[styles.profilepicture, {backgroundColor: colors.gray1}]} /> 
-      )
-    }
-
-    return (
+import DomainList from '../../../DiscoveryScreenV2/elements/DiscoveryItemList';
 
-        <View style={styles.card}>
-          <GlobalButton testID='pressbody' onPress={() => onPressBody(item)} style={styles.wrapProfile}>
-            {!isHashtag ? <React.Fragment>
-              {hanldeImage(item.image)}
-            </React.Fragment> : null}
-
-            <View style={styles.wrapTextProfile}>
-              <Text testID='name' style={styles.textProfileUsername}>
-                {isHashtag && "#"}{item.name || 'Anonymous'}
-              </Text>
-              <Text
-                testID='desc'
-                style={styles.textProfileFullName}
-                numberOfLines={1}
-                ellipsizeMode={'tail'}>
-                {item.description ? item.description : ''}
-              </Text>
-            </View>
-          </GlobalButton>
-          {item.isUnblocked ? (
-            <GlobalButton testID='isUnblock' onPress={handleSetBlock}>
-              <View style={styles.buttonBlockUser}>
-                <Text style={styles.textButtonBlockUser}>Block</Text>
-              </View>
-            </GlobalButton>
-          ) : (
-            <GlobalButton testID='isBlock'  onPress={handleSetUnblock}>
-              <View style={styles.buttonBlock}>
-                <Text style={styles.textButtonBlock}>Blocked</Text>
-              </View>
-            </GlobalButton>
-          )}
-        </View>
-    )
-}
+const BlockedList = (props) => {
+  const {item, isHashtag, handleSetBlock, handleSetUnblock, onPressBody} = props;
+  return (
+    <DomainList
+      key={`${item.blocked_action_id}`}
+      onPressBody={() => onPressBody(item)}
+      isBlockedSection={true}
+      handleSetBlock={() => handleSetBlock(item)}
+      handleSetUnblock={() => handleSetUnblock(item)}
+      isHashtag={isHashtag}
+      item={{
+        name: item.name,
+        image: item.image,
+        isUnblocked: item.isUnblocked,
+        description: item.description
+      }}
+    />
+  );
+};
 
 BlockedList.propTypes = {
-    item: PropTypes.object,
-    onPressList: PropTypes.func,
-    isHashtag: PropTypes.bool,
-    onPressBody: PropTypes.func
-}
+  item: PropTypes.object,
+  onPressBody: PropTypes.func,
+  handleSetBlock: PropTypes.func,
+  handleSetUnblock: PropTypes.func,
+  isHashtag: PropTypes.bool
+};
 
 BlockedList.defaultProps = {
-  onPressList: () => {},
   handleSetBlock: () => {},
   handleSetUnblock: () => {},
   onPressBody: () => {}
-}
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The refactoring of the `BlockedList` component seems to be well done. The code is cleaner and more maintainable by extracting the rendering logic into a separate `DomainList` component. This change also improves modularity as the `DomainList` component can now be reused in other parts of the application if needed. However, there's a minor issue with the prop types.

```diff
- BlockedList.propTypes = {
-    item: PropTypes.object,
-    onPressBody: PropTypes.func,
-    handleSetBlock: PropTypes.func,
-    handleSetUnblock: PropTypes.func,
-    isHashtag: PropTypes.bool
- };
+ BlockedList.propTypes = {
+    item: PropTypes.shape({
+      name: PropTypes.string,
+      image: PropTypes.string,
+      isUnblocked: PropTypes.bool,
+      description: PropTypes.string,
+      blocked_action_id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    }),
+    onPressBody: PropTypes.func.isRequired,
+    handleSetBlock: PropTypes.func.isRequired,
+    handleSetUnblock: PropTypes.func.isRequired,
+    isHashtag: PropTypes.bool
+ };
```
In the above diff, I've made the following changes:

1. Instead of using `PropTypes.object` for `item`, I've used `PropTypes.shape` to specify the shape of the object. This provides better type checking and makes it clear what properties are expected in the `item` object.
2. I've marked `onPressBody`, `handleSetBlock`, and `handleSetUnblock` as required using `isRequired`. Since these functions have default props, they will always be provided, so it's safe to mark them as required. This will help catch any issues where these props might not be provided due to a mistake.

Please consider making these changes to improve the robustness of your prop types checking.


<!-- This is an auto-generated comment by OSS CodeRabbit -->